src/fabric.c

```diff
@@ -507,6 +507,31 @@ void DEFAULT_SYMVER_PRE(fi_freeinfo)(struct fi_info *info)
 			free(info->fabric_attr->prov_name);
 			free(info->fabric_attr);
 		}
+		if (info->handle && info->handle->fclass == FI_CLASS_NIC) {
+			struct fid_nic *nic = (struct fid_nic*) info->handle;
+			if (nic->device_attr) {
+				free(nic->device_attr->name);
+				free(nic->device_attr->device_id);
+				free(nic->device_attr->device_version);
+				free(nic->device_attr->vendor_id);
+				free(nic->device_attr->driver);
+				free(nic->device_attr->firmware);
+				free(nic->device_attr);
+			}
+			free(nic->bus_attr);
+			if (nic->link_attr) {
+				free(nic->link_attr->address);
+				free(nic->link_attr->network_type);
+				free(nic->link_attr);
+			}
+			if (nic->prov_attr) {
+				// JMS How do we free this?  Its
+				// contents is only know to the
+				// provider.
+				free(nic->prov_attr);
+			}
+			free(nic);
+		}
```

Comment:
Can we move this to a separate function?  I haven't looked and don't remember, but I'm guessing fi_dupinfo needs to be updated as well.

Reply:
Moved to a subroutine.  Also added handling for `fi_dupinfo()`.